src/Arcus.WebApi.Logging.AzureFunctions/AzureFunctionsRequestTrackingMiddleware.cs

```diff
@@ -0,0 +1,160 @@
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Azure.Functions.Worker.Middleware;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Primitives;
+
+namespace Arcus.WebApi.Logging.AzureFunctions
+{
+    /// <summary>
+    /// Represents a middleware component that tracks a HTTP request in an Azure Functions environment.
+    /// </summary>
+    public class AzureFunctionsRequestTrackingMiddleware : RequestTrackingTemplate, IFunctionsWorkerMiddleware
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AzureFunctionsRequestTrackingMiddleware" /> class.
+        /// </summary>
+        public AzureFunctionsRequestTrackingMiddleware(RequestTrackingOptions options) : base(options)
+        {
+        }
+
+        /// <summary>
+        /// Invokes the middleware.
+        /// </summary>
+        /// <param name="context">The <see cref="T:Microsoft.Azure.Functions.Worker.FunctionContext" /> for the current invocation.</param>
+        /// <param name="next">The next middleware in the pipeline.</param>
+        /// <returns>A <see cref="T:System.Threading.Tasks.Task" /> that represents the asynchronous invocation.</returns>
+        public async Task Invoke(FunctionContext context, FunctionExecutionDelegate next)
+        {
+            ILogger logger = context.GetLogger<AzureFunctionsRequestTrackingMiddleware>();
+
+            HttpRequestData request = await context.GetHttpRequestDataAsync();
+            if (request is null || IsRequestPathOmitted(PathString.FromUriComponent(request.Url), logger))
+            {
+                await next(context);
+            }
+            else
+            {
+                if (Options.IncludeRequestBody)
+                {
+                    request = await EnableHttpRequestBuffering(context);
+                }
+
+                string requestBody = await GetPotentialRequestBodyAsync(request, logger);
+                using (var measurement = DurationMeasurement.Start())
+                {
+                    try
+                    {
+                        await next(context);
+                    }
+                    finally
+                    {
+                        HttpResponseData response = context.GetHttpResponseData();
+                        var attributeTrackedStatusCodes = Enumerable.Empty<StatusCodeRange>();
+
+                        if (response != null && AllowedToTrackStatusCode((int) response.StatusCode, attributeTrackedStatusCodes, logger))
+                        {
+                            string responseBody = await GetPotentialResponseBodyAsync(response, logger);
+                            LogRequest(requestBody, responseBody, request, response, measurement, logger); 
+                        }
+                    } 
+                }
+            }
+        }
+
+        private static async Task<HttpRequestData> EnableHttpRequestBuffering(FunctionContext context)
```

Comment:
It is an `async` method, so we should suffix it with `Async`

Reply:
Woops, you're right.

Reply:
https://github.com/arcus-azure/arcus.webapi/pull/383/commits/96b8a9e4e1edc33f0b856b7a614ca5a023df9d26